lib.nix

```diff
@@ -1,18 +1,42 @@
-let
-  # iohk-nix can be overridden for debugging purposes by setting
-  # NIX_PATH=iohk_nix=/path/to/iohk-nix
-  iohkNix = import (
-    let try = builtins.tryEval <iohk_nix>;
-    in if try.success
-    then builtins.trace "using host <iohk_nix>" try.value
-    else
-      let
-        spec = builtins.fromJSON (builtins.readFile ./iohk-nix.json);
-      in builtins.fetchTarball {
-        url = "${spec.url}/archive/${spec.rev}.tar.gz";
-        inherit (spec) sha256;
-      }) {};
+{ system ? builtins.currentSystem
+, crossSystem ? null
+, config ? {}
+, overlays ? []
+}:
 
-  pkgs = iohkNix.pkgs;
+let
+  sources = import ./nix/sources.nix;
+  iohkNix = import sources.iohk-nix {
+    sourcesOverride = sources;
+  };
+  haskellNix = import sources."haskell.nix";
+  args = haskellNix // {
+    inherit system crossSystem;
+    overlays = (haskellNix.overlays or []) ++ overlays;
+    config = (haskellNix.config or {}) // config;
+  };
+  nixpkgs = import sources.nixpkgs;
+  pkgs = nixpkgs args;
+  haskellPackages = import ./nix/pkgs.nix {
+    inherit pkgs;
+    src = ./.;
+  };
   lib = pkgs.lib;
-in lib // { inherit iohkNix pkgs; inherit (iohkNix) nix-tools; }
+  niv = (import sources.niv {}).niv;
+  isCardanoBase = with lib; package:
+    (package.isHaskell or false) &&
+      (hasPrefix "cardano-base" package.identifier.name);
```

Comment:
need to fix the prefixes here. none of these packages start with cardano-base

Reply:
Prefix removed in commit 729da10